pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -185,6 +197,36 @@ func (i *Gatherer) Gather(ctx context.Context, recorder record.Interface) error
 	)
 }
 
+func handlePendingPodsNetworkEvents(pods []corev1.Pod, eventsGetter corev1client.EventsGetter) ([]corev1.Event, error) {
+	filteredEvents := []corev1.Event{}
+	if len(pods) == 0 {
+		return filteredEvents, nil
+	}
+	podNamespaces := sets.NewString()
+	for _, pod := range pods {
+		if podNamespaces.Has(pod.Namespace) {
+			continue
+		}
+		podNamespaces.Insert(pod.Namespace)
+	}
+	for _, namespace := range podNamespaces.List() {
+		namespaceEvents, err := eventsGetter.Events(namespace).List(metav1.ListOptions{})
```

Comment:
We probably need to bound the runtime of this call in the future - i.e. if it hangs it hangs the gather loop.  That does not have to be done in this PR.